### PR TITLE
Primary rail

### PR DESCRIPTION
**Pull request by KCraw**
This includes #26 

After playing with this on a device, I added back in a calculation for primary rail axis during the grant stage using both delta and velocity. Only takes control if you are "more" X-axis than Y-axis by both measures.

### CODE REVIEW

File: components/app/index.js [hunk 1]
```diff
@@ -27,56 +27,44 @@ export default class Gustave extends Component {
...
-    let recommendations = this.state.recommendations.filter(rec => recommendation.id !== rec.id);
-    this.setState({
-      saved,
-      justSaved: true,
-      recommendations,
-    });
+  onSaveRecommendation(recommendationId) {
+    database.saveUserRecommendation(this.state.user.id, recommendationId);
+    this.setState({justSaved: true});
     this.setState({justSaved: false});
```
Comment (jimbol) [inline components/app/index.js]: There a reason we're setting `justSaved` twice?

File: components/app/index.js [hunk 2]
```diff
@@ -113,7 +101,7 @@ export default class Gustave extends Component {
           <RecommendationsScene
             style={styles.scene}
             isLoadingMore={this.state.isLoadingMore}
-            recommendations={this.state.recommendations}
+            nextRecommendation={database.getUserRecommendations(this.state.user.id)[0]}
```
Comment (jimbol) [inline components/app/index.js]: `nextRecommendation` is going to be harder to determine when we have a tree of results.  This works for now though.

File: components/app/index.js (hunk 1 above)
Comment (KCraw) [inline components/app/index.js]: As a temp hack to trigger the twinkle twinkle of the little star

File: components/app/index.js (hunk 2 above)
Comment (KCraw) [inline components/app/index.js]: It wont be determined by this component at all. This only exists so we can manually control rerender on recommendation change. 

Perhaps my naming convention should be tweaked if this isnt clear. 

this.props.recommendation 
this.state.shownRecommendation

Which are only ever (temporarily) out of sync if props.recommendation changes while the user is looking at a recommendation.

File: components/app/index.js (hunk 1 above)
Comment (jimbol) [inline components/app/index.js]: Ah! I see!  Better leave a comment or wrap it in a named method so we know in the future.
```
twinkleTwinkleLittleStar(){
  this.setState({justSaved: true});
  this.setState({justSaved: false});
}
```

Comment (KCraw) [inline components/app/index.js]: 👍

**Comment by jimbol** [comment]
snagging these changes

**Pull request merged**